bp_fe/src/include/bp_fe_pc_gen.vh

```diff
@@ -0,0 +1,208 @@
+/*
+ * pc_gen.vh
+ *
+ * pc_gen.vh provides the structs for PC generation logics. PC generation use
+ * previous PCs to determine the next PC. PC generation also inspects instrcutions
+ * and flag RVC if the instruction is compressed.  
+ * 
+ * PC generation consists of instruction scanning and Branch Prediction (BP). The
+ * instruction scanning first scans the control flow instructions, to determines
+ * whether the instruction is compressed or not. If the instruction is compressed,
+ * the pc_gen will flag the categories of the instruction to other modules in
+ * backend.  BP uses the previous PC to look up the corresponding entry in the
+ * Branch History Table (BHT).  If the BHT entry predicts taken, BP find the
+ * correpsonding entry in the Branch Target Buffer (BTB) as the next PC.  In the
+ * case of call-ret pairs, BP pushes the call target to Return Address Stack
+ * (RAS) and pops the address in the ret instruction.
+*/
+
+`ifndef BP_FE_PC_GEN_VH
+`define BP_FE_PC_GEN_VH
+
+`ifndef BSG_DEFINES_V
+`define BSG_DEFINES_V
+`include "bsg_defines.v"
+`endif
+
+`ifndef BP_COMMON_FE_BE_IF_VH
+`define BP_COMMON_FE_BE_IF_VH
+`include "bp_common_fe_be_if.vh"
+`endif
+
+// import bp_common_pkg::*;
+import pc_gen_pkg::*;
```

Comment:
don't import at this level.  If you `include _in_ the package, then the package doesn't get globally defined